cmd/kubectl-gadget/version.go

```diff
@@ -66,12 +81,27 @@ var versionCmd = &cobra.Command{
 			// Exactly one running gadget instance found, use it
 			runtimeGlobalParams.Set(grpcruntime.ParamGadgetNamespace, gadgetNamespaces[0])
 			info, err := grpcRuntime.InitDeployInfo()
+
 			if err != nil {
 				fmt.Fprintf(os.Stderr, "Error: loading deploy info: %s\n", err)
 			} else {
-				versionInfo.ServerVersion = &Version{
-					Version: info.ServerVersion,
-				}
+				serverVersionStr := strings.TrimPrefix(info.ServerVersion, "v")
+                versionParts := strings.Split(serverVersionStr, ".")
+                serverMajor := "0"
+                serverMinor := "0"
+                if len(versionParts) >= 2 {
+                    serverMajor = versionParts[0]
+                    serverMinor = versionParts[1]
+                }
+
+                versionInfo.ServerVersion = &Version{
+                    Major:      serverMajor,
+                    Minor:      serverMinor,
+                    GitVersion: fmt.Sprintf("v%s", serverVersionStr),
+                    GoVersion:  runtime.Version(),
+                    Compiler:   runtime.Compiler,
+                    Platform:   fmt.Sprintf("%s/%s", runtime.GOOS, runtime.GOARCH),
```

Comment:
`runtime.*` are not coming from the server here. I think you need to update the grpc method `GetInfo` (`pkg/gadget-service/api/api.proto`) to include the necessary information.